modules/diskq/qdisk.c

```diff
@@ -461,12 +461,6 @@ _try_reading_record_length(QDisk *self, guint32 *record_length)
 {
   guint32 read_record_length;
   gssize bytes_read = _read_record_length_from_disk(self, &read_record_length);
-  if (bytes_read == 0)
-    {
-      /* hmm, we are either at EOF or at hdr->qout_ofs, we need to wrap */
-      self->hdr->read_head = QDISK_RESERVED_SPACE;
-      bytes_read = _read_record_length_from_disk(self, &read_record_length);
```

Comment:
I don't understand the reasoning in the commit message. You mentioned 62152b1, but that was an equivalent transformation.

Are you sure this is dead code? I mean, for example, we don't update positions if either `_try_reading_record_length()` or `_read_record_from_disk()` fails.

Reply:
We discussed this one IRL, it is dead code.
In case any of the mentioned calls fail, that means the disk-buffer is corrupt, and we'll move the current disk-buffer and create a new one.

Reply:
On error cases, the disk-buffer file is replaced by a new disk-buffer file as it seems to be corrupted.

Reply:
On the other hand, could a truncate by `qdisk_push_tail` cause that read_head is at end of file?

Reply:
@gaborznagy 

> On the other hand, could a truncate by qdisk_push_tail cause that read_head is at end of file?

That cannot happen, as we only truncate in `qdisk_push_tail()` when `write_head > read_head`.

Reply:
We discussed and checked this: we don't see it possible that `read_head` would be at EOF position:
- during `push_tail` we can truncate to `write_head` if `read_head` follows `write_head` (read_head < write_head and not read_head <= write_head !!!). In this case even if it's truncated, read_head would have at least one message until EOF
- `qdisk_pop_head` will not be called on an empty disk-buffer: On LogQueueDisk level pop_head is not called if the queue is empty. This is important as it can happen that `qdisk_pop_head` doesn't set `read_head` to the file's beginning even if the queue is empty due to `truncate-size-ratio`.

However, if a read at EOF would still happen, the record_length reading will got a "short read" error, and the disk-buffer will be marked corrupted and a new one would be started.